Move helpers to torch.distributed.utils

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #92881

Move these helpers to torch.distributed.utils as they will be consumed
by DDP for native mixed precision implementation

Differential Revision: [D43245719](https://our.internmc.facebook.com/intern/diff/D43245719)

rohan-varma committed:
Move helpers to torch.distributed.utils

Move these helpers to torch.distributed.utils as they will be consumed
by DDP for native mixed precision implementation

Differential Revision: [D42674745](https://our.internmc.facebook.com/intern/diff/D42674745/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Move helpers to torch.distributed.utils"


Move these helpers to torch.distributed.utils as they will be consumed
by DDP for native mixed precision implementation

Differential Revision: [D42674745](https://our.internmc.facebook.com/intern/diff/D42674745/)

[ghstack-poisoned]

zhaojuanmao reviewed (approved):
codes look good, please fix the lint and make sure tests are green before landing

rohan-varma committed:
Update on "Move helpers to torch.distributed.utils"


Move these helpers to torch.distributed.utils as they will be consumed
by DDP for native mixed precision implementation

Differential Revision: [D42674745](https://our.internmc.facebook.com/intern/diff/D42674745/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Move helpers to torch.distributed.utils"


Move these helpers to torch.distributed.utils as they will be consumed
by DDP for native mixed precision implementation

Differential Revision: [D42674745](https://our.internmc.facebook.com/intern/diff/D42674745/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Move helpers to torch.distributed.utils"


Move these helpers to torch.distributed.utils as they will be consumed
by DDP for native mixed precision implementation

Differential Revision: [D42674745](https://our.internmc.facebook.com/intern/diff/D42674745/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Move helpers to torch.distributed.utils"


Move these helpers to torch.distributed.utils as they will be consumed
by DDP for native mixed precision implementation

Differential Revision: [D42674745](https://our.internmc.facebook.com/intern/diff/D42674745/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Move helpers to torch.distributed.utils"


Move these helpers to torch.distributed.utils as they will be consumed
by DDP for native mixed precision implementation

Differential Revision: [D42674745](https://our.internmc.facebook.com/intern/diff/D42674745/)

[ghstack-poisoned]

rohan-varma commented:
@pytorchbot merge -f "failure unrelated"

rohan-varma committed:
Update on "Move helpers to torch.distributed.utils"


Move these helpers to torch.distributed.utils as they will be consumed
by DDP for native mixed precision implementation

Differential Revision: [D43245719](https://our.internmc.facebook.com/intern/diff/D43245719)

[ghstack-poisoned]
